### PR TITLE
Many Edit payouts bugs

### PR DESCRIPTION
**Pull request by johnnyd-eth**
## What does this PR do and why?

Fixes JB-90
Fixes JB-87
Fixes JB-86 
Fixes JB-85

I've tested the create flow and found no regressions.

Follow-up issues:
JB-91
JB-93

## Screenshots or screen recordings

https://user-images.githubusercontent.com/96150256/224208645-8a130c72-a74e-4777-9115-01b137b5df28.mp4

**/v2/p/553 demo:**

https://user-images.githubusercontent.com/96150256/224235929-b01101fe-cf96-4b0e-bd6c-6572cdb7b2b1.mp4

## Acceptance checklist

- [ ] I have evaluated the [Approval Guidelines](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#approval-guidelines) for this PR.
- [ ] I have tested this PR in [all supported browsers](https://github.com/jbx-protocol/juice-interface/blob/main/CONTRIBUTING.md#supported-browsers).
- [ ] I have tested this PR in dark mode and light mode (if applicable).

### CODE REVIEW

File: src/components/Allocation/AddEditAllocationModal.tsx [hunk 1]
```diff
@@ -132,7 +132,7 @@ export const AddEditAllocationModal = ({
       result = {
         projectOwner: false,
         beneficiary: fields.address,
-        projectId: fields.juiceboxProjectId,
+        projectId: fields.juiceboxProjectId, //BigNumber.from(fields.juiceboxProjectId ?? 0).toHexString(),
```
Comment (tomquirk) [inline src/components/Allocation/AddEditAllocationModal.tsx:135]: what the go here

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/DistributionSplitModal.tsx [hunk 2]
```diff
@@ -289,7 +289,7 @@ export function DistributionSplitModal({
 
   return (
     <Modal
-      title={mode === 'Edit' ? t`Edit payout` : t`Add new payout`}
+      title={mode === 'Edit' ? t`Edit payout` : t`Add new payoutX`}
```
Comment (tomquirk) [inline src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/DistributionSplitModal.tsx:292]: typo

File: src/components/Allocation/AllocationList.tsx [hunk 3]
```diff
@@ -260,10 +268,11 @@ const entityToAllocation = (
   entity: AddEditAllocationModalEntity & { projectOwner: false },
   totalAllocationAmount: BigNumber | undefined,
 ): AllocationSplit => {
+  const projectIdHex = projectIdToHex(entity.projectId)
   const allocationProps = {
-    id: allocationId(entity.beneficiary!, entity.projectId),
+    id: allocationId(entity.beneficiary!, projectIdHex),
     beneficiary: entity.beneficiary,
-    projectId: entity.projectId,
+    projectId: projectIdHex,
```
Comment (tomquirk) [inline src/components/Allocation/AllocationList.tsx:275]: why do project ids need to be hexed?

File: src/components/Allocation/AllocationList.tsx [hunk 4]
```diff
@@ -17,8 +18,13 @@ import { Allocation, AllocationSplit } from './Allocation'
 
 type AddButtonSize = 'small' | 'large'
 
-const allocationId = (beneficiary: string, projectId: string | undefined) =>
-  `${beneficiary}${projectId ? `-${projectId}` : ''}`
+const allocationId = (
+  beneficiary: string,
+  projectIdHex: string | undefined,
```
Comment (tomquirk) [inline src/components/Allocation/AllocationList.tsx:23]: why does this need to be hex?

File: src/components/Allocation/AllocationList.tsx (hunk 3 above)
Comment (johnnyd-eth) [reply src/components/Allocation/AllocationList.tsx:275]: https://discord.com/channels/939317843059679252/1080714793595441182/1083628114179014696

See JB-93

File: src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/DistributionSplitModal.tsx (hunk 2 above)
Comment (johnnyd-eth) [reply src/components/v2v3/shared/FundingCycleConfigurationDrawers/DistributionSplitModal/DistributionSplitModal.tsx:292]: Apologies

File: src/components/Allocation/AddEditAllocationModal.tsx (hunk 1 above)
Comment (johnnyd-eth) [reply src/components/Allocation/AddEditAllocationModal.tsx:135]: Double apologies

**Comment by wraeth-eth** [comment]
Regression in create - Can add the same wallet address as many times as you want
image

**Comment by wraeth-eth** [comment]
Another regression in create - makes a bad rounding error when editing some numbers close to ~99.999999999

https://user-images.githubusercontent.com/104132113/224522884-d7988dbc-5c43-4f5e-8cad-2b584af8341e.mov

**Comment by wraeth-eth** [comment]
> Another regression in create - makes a bad rounding error when editing some numbers close to ~99.999999999

Nevermind @johnnyd-eth , this one actually exists on main currently

**Comment by wraeth-eth** [comment]
Another similar regression, you can add duplicate project id payments to the same beneficiary. i.e
Payout 1: {beneficiary: wraeth.eth, projectId: 1}
Payout 2: {beneficiary: wraeth.eth, projectId: 1}

Similar to the regression mentioned [here](https://github.com/jbx-protocol/juice-interface/pull/3189#issuecomment-1465083677).

It looks like this is also related to the discrepency between comparing integers and hex numbers. The already added payout is expressed in hex (0x01), whereas the other is 1

**Comment by johnnyd-eth** [comment]
> Another similar regression, you can add duplicate project id payments to the same beneficiary. i.e Payout 1: {beneficiary: wraeth.eth, projectId: 1} Payout 2: {beneficiary: wraeth.eth, projectId: 1}
> 
> Similar to the regression mentioned [here](https://github.com/jbx-protocol/juice-interface/pull/3189#issuecomment-1465083677).
> 
> It looks like this is also related to the discrepancy between comparing integers and hex numbers. The already added payout is expressed in hex (0x01), whereas the other is 1

https://user-images.githubusercontent.com/96150256/224549289-84817790-e566-4b10-b8f4-c850d635c142.mp4

Fixed both these issues sir @wraeth-eth. Thanks for that catch!

**Review by wraeth-eth — APPROVED** [review]
Full send it @johnnyd-eth

**Pull request merged**